core/arch/arm/kernel/kern.ld.S

```diff
@@ -50,13 +50,19 @@
 
 #include <platform_config.h>
 
+#ifndef SMALL_PAGE_SIZE
+#define SMALL_PAGE_SIZE		4096
+#endif
+
 OUTPUT_FORMAT(CFG_KERN_LINKER_FORMAT)
 OUTPUT_ARCH(CFG_KERN_LINKER_ARCH)
 
 ENTRY(_start)
 SECTIONS
 {
 	. = CFG_TEE_LOAD_ADDR;
+	. = ALIGN(SMALL_PAGE_SIZE);
+	__core_unpg_rx_start = .;
```

Comment:
I'm basically OK with having redundant definitions as it helps understanding, but definitions that are supposed to be the same should be assigned at the same place. I think the new locations are better so please move `__text_start` and friends.